augmentation/augmentation.hpp

```diff
@@ -0,0 +1,172 @@
+/**
+ * @file augmentation.hpp
+ * @author Kartik Dutt
+ * 
+ * Definition of Augmentation class for augmenting data.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#include <mlpack/methods/ann/layer/bilinear_interpolation.hpp>
+#include <boost/regex.hpp>
+
+#ifndef MODELS_AUGMENTATION_HPP
+#define MODELS_AUGMENTATION_HPP
+
+/**
+ * Augmentation class used to perform augmentations / transform the data.
+ * For the list of supported augmentation, take a look at our wiki page.
+ *
+ * @code
+ * Augmentation<> augmentation({"horizontal-flip", "resize = (224, 224)"}, 0.2);
+ * augmentation.Transform(dataloader.TrainFeatures);
+ * @endcode
+ * 
+ * @tparam DatasetType Datatype on which augmentation will be done.
+ */
+template<typename DatasetType = arma::mat>
```

Comment:
For follow-up with @zoq [comment](https://github.com/mlpack/models/pull/13/#discussion_r436166401), can you remove this template parameter and instantiate it locally?

Reply:
Done, I'll make one more commit tonight or tomorrow morning and then it will be ready to go from my side too. Regards.

Reply:
This is ready to go from my side but the matrix type gives invalid read sometimes. I will try to figure out where I am making an invalid read.

Reply:
Figured it out, there was a type between width and height. Fixed now. This is ready for review. Kindly let me know what you think.